File: core-examples/src/main/rb/io/vertx/example/core/http/sendfile/sendfile.rb

# In reality it's highly recommend you use Apex for applications like this.

$vertx.create_http_server().request_handler() { |req|
  filename = nil
  if (req.path().equals("/"))
    filename = "index.html"
  elsif (req.path().equals("/page1.html"))
    filename = "page1.html"
  elsif (req.path().equals("/page2.html"))
    filename = "page2.html"
  else
    req.response().set_status_code(404).end()
  end
  if (filename != nil)
    req.response().send_file(filename)
  end
}.listen(8080)

File: core-examples/src/main/js/io/vertx/example/core/http/sendfile/sendfile.js

// In reality it's highly recommend you use Apex for applications like this.

vertx.createHttpServer().requestHandler(function (req) {
  var filename = null;
  if (req.path() == "/") {
    filename = "index.html";
  } else if (req.path() == "/page1.html") {
    filename = "page1.html";
  } else if (req.path() == "/page2.html") {
    filename = "page2.html";
  } else {
    req.response().setStatusCode(404).end();
  };
  if (filename !== null) {
    req.response().sendFile(filename);
  };
}).listen(8080);
